subworkflows/local/prepare_genome/main.nf

```diff
@@ -0,0 +1,149 @@
+    def files = files_path.listFiles()
+    if (files == null || files.length == 0) {
+        throw new Exception("The provided bowtie_index path doesn't contain any files.")
+    }
+    def index_prefix = ''
+    for (file_path in files) {
+        def file_name = file_path.getName()
+        if (file_name.endsWith(".1.ebwt") && !file_name.endsWith(".rev.1.ebwt")) {
+            index_prefix = file_name.substring(0, file_name.lastIndexOf(".1.ebwt"))
+            break
+        }
+    }
+    if (index_prefix == '') {
+        throw new Exception("Unable to extract the prefix from the Bowtie index files. No file with the '.1.ebwt' extension was found. Please ensure that the correct files are in the specified path.")
+    }
+    return index_prefix
+}
+
+
+workflow PREPARE_GENOME {
+    take:
+    val_fasta                      // file: /path/to/genome.fasta
+    val_bowtie_index               // file or directory: /path/to/bowtie/ or /path/to/bowtie.tar.gz
+    val_mirtrace_species           // string: Species for miRTrace
+    val_rrna                       // string: Path to the rRNA fasta file to be used as contamination database.
+    val_trna                       // string: Path to the tRNA fasta file to be used as contamination database.
+    val_cdna                       // string: Path to the cDNA fasta file to be used as contamination database.
+    val_ncrna                      // string: Path to the ncRNA fasta file to be used as contamination database.
+    val_pirna                      // string: Path to the piRNA fasta file to be used as contamination database.
+    val_other_contamination        // string: Path to the additional fasta file to be used as contamination database.
+    val_fastp_known_mirna_adapters // string: Path to Fasta with known miRNA adapter sequences for adapter trimming
+    val_mirna_gtf                  // string: Path to GFF/GTF file with coordinates positions of precursor and miRNAs
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Parameter channel handling
+    ch_fasta                  = val_fasta                     ? Channel.fromPath(val_fasta, checkIfExists: true).map{ it -> [ [id:it.baseName], it ] }.collect()  : Channel.empty()
+    ch_bowtie_index           = val_bowtie_index              ? Channel.fromPath(val_bowtie_index, checkIfExists: true).map{ it -> [ [id:""], it ] }.collect()    : Channel.empty()
+
+    bool_mirtrace_species     = val_mirtrace_species          ? true : false
+    bool_has_fasta            = val_fasta                     ? true : false
+
+    ch_mirtrace_species       = val_mirtrace_species          ? Channel.value(val_mirtrace_species) : Channel.empty()
+    mirna_gtf_from_species    = val_mirtrace_species          ? (val_mirtrace_species == 'hsa' ? "https://raw.githubusercontent.com/nf-core/test-datasets/smrnaseq/reference/hsa.gff3" : "https://mirbase.org/download/CURRENT/genomes/${val_mirtrace_species}.gff3") : false
```

Comment:
You should place the URL as a parameter and insert that here rather than having a hidden URL. This makes it clearer to users they need to provide a local path if e.g. on an online cluster here - otherwise an implicit download could confuse an user.
```suggestion
    mirna_gtf_from_species    = val_mirtrace_species          ? (val_mirtrace_species == 'hsa' ? params.hsa_ref : "https://mirbase.org/download/CURRENT/genomes/${val_mirtrace_species}.gff3") : false
```

Or something like that